### PR TITLE
HTTPS, SSH sessions, refactoring

### LINKED ISSUE #42: Inertia SSH Command
Add a command `inertia SSH [REMOTE]` that establishes an interactive session with your remote (just a wrapper around `ssh user@ip`).

### LINKED ISSUE #59: Secure client-server communication
Right now the daemon sets up an HTTP (not HTTPS) listener. This makes the token authorization system vulnerable to replay attacks since the token is sent in the clear.

To fix: Use HTTPS exclusively. Maybe there's a way to generate a cert when creating a new remote, and informing the client-side to trust that cert?

### LINKED ISSUE #121: Multi-remote deployment commands not identified correctly

### PR DESCRIPTION
**Pull request by bobheadxi**
:tickets: **Ticket(s)**: 
- Closes #59 
- Closes #42 
- Fixes #121

---

Hoping to deploy `v0.1.1` after this PR for the [ubclaunchpad/bumper](https://github.com/ubclaunchpad/bumper) team
## :construction_worker: Changes

- SSL certificate generation and HTTPS communications. This certificate is self-signed, however, and will raise a warning - should we worry about this? 🤔  See previous POC PR for old discussions at #63 
screen shot 2018-03-17 at 3 09 36 pm

- Start a SSH session with your remote using Inertia!

- Squashed some bugs, improved `init` feedback, and refactored `client` for clarity (mostly separating classes out better)

## :flashlight: Testing Instructions

```bash
inertia $DEPLOYMENT init
inertia $DEPLOYMENT up # will be done over https
```

And:
```bash
bobbook:inertia-deploy-test robertlin$ inertia tweet ssh
Linux tweetlibs 4.9.0-5-amd64 #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Sat Mar 17 01:30:58 2018 from 24.86.161.145
robertlin@tweetlibs:~$ ls
certs  project  ssl  tls  tls-linux
```

### CODE REVIEW

File: client/remote.go [hunk 1]
```diff
@@ -28,88 +26,69 @@ type DaemonConfig struct {
...
+	println("=============================\n")
 
 	// Output deploy key to user.
-	println("GitHub Deploy Key (add here https://www.github.com/<your_repo>/settings/keys/new): ")
+	println("GitHub Deploy Key (add to https://www.github.com/<your_repo>/settings/keys/new): ")
 	println(pub.String())
 
 	// Output Webhook url to user.
-	println("GitHub WebHook URL (add here https://www.github.com/<your_repo>/settings/hooks/new): ")
+	println("GitHub WebHook URL (add to https://www.github.com/<your_repo>/settings/hooks/new): ")
```
Comment (chadlagore) [inline client/remote.go]: Might be nice to backfill the repo name here (if that’s an option now)

File: daemon/server.go [hunk 2]
```diff
@@ -4,17 +4,26 @@ import (
...
+var daemonVersion string
+
 const (
 	// specify location of deployed project
 	projectDirectory = "/app/host/project"
 
+	// specify location of SSL certificate
+	sslDirectory  = "/app/host/ssl/"
+	daemonSSLCert = sslDirectory + "daemon.cert"
+	daemonSSLKey  = sslDirectory + "daemon.key"
```
Comment (chadlagore) [inline daemon/server.go]: Might as well use Join

File: daemon/certificate.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+		priv, err = ecdsa.GenerateKey(elliptic.P224(), rand.Reader)
+	case "P256":
+		priv, err = ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	case "P384":
+		priv, err = ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	case "P521":
+		priv, err = ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+	default:
+		fmt.Fprintf(os.Stderr, "Unrecognized elliptic curve: %q", ecdsaCurve)
+		os.Exit(1)
```
Comment (chadlagore) [inline daemon/certificate.go]: I take it this is their code? Maybe we want to change the error handling to propagate up like we do elsewhere?

File: client/remote.go (hunk 1 above)
Comment (bobheadxi) [reply client/remote.go]: #123

File: daemon/certificate.go (hunk 3 above)
Comment (bobheadxi) [reply daemon/certificate.go]: Good point, just improved this 👍

File: daemon/server.go (hunk 2 above)
Comment (bobheadxi) [reply daemon/server.go]: Don't think we can if we want to keep these as `const` 🤔

**Pull request merged**